pydis_site/apps/content/resources/guides/python-guides/subclassing_context.md

```diff
@@ -0,0 +1,179 @@
+
+## Basic Subclassing
+First, the [documentation on inheritance](https://docs.python.org/3/tutorial/classes.html#inheritance) on subclassing will provide some fundamental knowledge, which is highly suggested before moving on to this topic, as subclassing [Context](https://discordpy.readthedocs.io/en/latest/ext/commands/api.html#discord.ext.commands.Context) can ultimately be a complicated task.
+
+## The benefits of subclassing Context
+Subclassing Context can be very beneficial as it allows you to set custom Context methods that can be used in your code. Some applications and examples are provided below.
+
+In order to subclass Context, you are required to subclass `commands.Bot` as well. More on subclassing Bot can be read here [link placeholder] but here is a short example.
+
+```py
+# This View is for the prompt function below in the Context subclass.
+# This is completely optional and its purpose is demonstrating applications of subclassing Context.
+
+
+class PromptView(discord.ui.View):
+    def __init__(
+        self,
+        *,
+        timeout: float,
+        author_id: int,
+        ctx: commands.Context,
+        delete_after: bool,
+    ) -> None:
+        super().__init__(timeout=timeout)
+        self.value: typing.Optional[bool] = None
+        self.delete_after: bool = delete_after
+        self.author_id: int = author_id
+        self.ctx: Context = ctx
+        self.message: typing.Optional[discord.Message] = None
+        
+        
+    '''Ensure that other members do not confirm or deny'''
+    async def interaction_check(self, interaction: discord.Interaction) -> bool:
+        if interaction.user and interaction.user.id == self.author_id:
+            return True
+        else:
+            await interaction.response.send_message(
+                "This confirmation dialog is not for you.", ephemeral=True
+            )
+            return False
+
+    async def on_timeout(self) -> None:
+        if self.delete_after and self.message:
+            await self.message.delete()
+        # delete the message if there is no response
+
+    @discord.ui.button(label="Confirm", style=discord.ButtonStyle.green)
+    async def confirm(
+        self, button: discord.ui.Button, interaction: discord.Interaction
+    ):
+        self.value = True  # returns True if the confirm button is pressed
+        await interaction.response.defer()
+        if self.delete_after:
+            await interaction.delete_original_message()
+        self.stop()  # stops the view and deletes the confirmation message
+
+    @discord.ui.button(label="Cancel", style=discord.ButtonStyle.red)
+    async def cancel(self, button: discord.ui.Button, interaction: discord.Interaction):
+        self.value = False  # returns False if the cancel button is pressed
+        await interaction.response.defer()
+        if self.delete_after:
+            await interaction.delete_original_message()
+        self.stop()  # stops the view and deletes the confirmation
```

Comment:
The focus is meant to be on the context, yet this is placed first and is probably longer than the actual context subclass.

Can we consider another type of example?